src/saltext/vmware/states/esxi.py

```diff
@@ -1,3 +1,216 @@
+    for host in diff:
+        if __opts__["test"]:
+            ret[
+                "comment"
+            ] = "User {} will be created on {} host(s) and updated on {} host(s).".format(
+                name, create, update
+            )
+            ret["result"] = None
+        elif diff[host]["action"] == "update":
+            __salt__["vmware_esxi.update_user"](
+                user_name=name,
+                password=password,
+                description=description,
+                datacenter_name=datacenter_name,
+                cluster_name=cluster_name,
+                host_name=host,
+                service_instance=service_instance,
+            )
+            ret["comment"] = "User {} created on {} host(s) and updated on {} host(s).".format(
+                name, create, update
+            )
+            ret["changes"] = diff
+            ret["result"] = True
+        else:
+            __salt__["vmware_esxi.add_user"](
+                user_name=name,
+                password=password,
+                description=description,
+                datacenter_name=datacenter_name,
+                cluster_name=cluster_name,
+                host_name=host,
+                service_instance=service_instance,
+            )
+            ret["comment"] = "User {} created on {} host(s) and updated on {} host(s).".format(
+                name, create, update
+            )
+            ret["changes"] = diff
+            ret["result"] = True
+    return ret
+
+
+def user_absent(
+    name,
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Remove local user on matching ESXi hosts.
+
+    name
+        User to create on matching ESXi hosts. (required).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: yaml
+
+    Remove User:
+      vmware_esxi.user_absent:
+        - name: local_user
+
+    """
+    log.debug("Running vmware_esxi.user_absent")
+    ret = {"name": name, "result": None, "comment": "", "changes": {}}
+    delete = 0
+    diff = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    user = __salt__["vmware_esxi.get_user"](
+        user_name=name,
+        datacenter_name=datacenter_name,
+        cluster_name=cluster_name,
+        host_name=host_name,
+        service_instance=service_instance,
+    )
+    hosts = __salt__["vmware_esxi.list_hosts"](
+        datacenter_name=datacenter_name,
+        cluster_name=cluster_name,
+        host_name=host_name,
+        service_instance=service_instance,
+    )
+    for host in hosts:
+        if host in user:
```

Comment:
same as above, it's just reads like a bug at first glance

Reply:
sure will change the name of the var